[DOCS-4821] Track RUM usage with usage attribution tags (#16823)
DataDog/documentation: content/en/real_user_monitoring/guide/tracking-rum-usage-with-usage-attribution-tags.md

---
title: Track RUM Usage with Usage Attribution Tags
kind: guide
beta: true
description: Learn how to track RUM usage with custom attribution tags
aliases:
- /real_user_monitoring/guide/track-rum-usage-with-attribution-tags/
further_reading:
- link: '/account_management/billing/usage_attribution/'
  tag: 'Documentation'
  text: 'Plan and Usage Settings'
---

## Overview

The [Usage Attribution][1] page provides information and functionality related to data usage and usage types. By default, the data usage can be viewed and filtered by broader categories such as product, organization, or tag keys. This guide describes how to configure RUM usage attribution so it can be viewed by custom categories on the Usage Attribution page (accurate to +/- 20% of the actual value). This can help you track RUM sessions and costs for different departments, products, or other categories, instead of viewing a single aggregate number. 

> As an example, we'll walk through how to track RUM usage by department.

Reviewer: ```suggestion
As an example, this guide walks through how to track RUM usage by department.
```

## Set up RUM usage attribution

### Check your tags

Categories for usage are determined by tags. Before setting up your RUM usage attribution, make sure that the tags you want to use are configured on the Usage Attribution page. Click **Edit tags**, then select the tags that you want to use to view usage and click **Save**. In this example, we've added "department" as a tag.

{{< img src="real_user_monitoring/guide/rum-usage-attribution-tags/rum-use-attribution-tags-1.jpeg" alt="Check your tags on the Usage Attribution page" style="width:100%;">}}

### Add tags to your RUM sessions
Once your usage attribution tags have been configured, you can tag your RUM sessions with them. 

To set tags for **browser sessions**, set the RUM global context at the start of the session (right after calling `datadogRum.init`) using the [`setGlobalContextProperty`][2] method. For example, here's how we would tag sessions so they can be tracked for the marketing department: 

```javascript
datadogRum.setGlobalContextProperty('department', 'marketing');
```

To set tags for **mobile sessions**, use the [`addAttribute`][5] method. Here's an example:

```javascript
//Android

> GlobalRum.addAttribute(‘department’, ‘marketing’)

Reviewer: ```suggestion
GlobalRum.addAttribute('department', 'marketing')
```

//iOS

> Global.rum.addAttribute(forKey: ‘department’, value: ‘marketing’)

Reviewer: ```suggestion
Global.rum.addAttribute(forKey: 'department', value: 'marketing')
```

```

> **Note**: We include a few tags by default (service, env, version, and application.id). For anything else, set the global context using the method above.

Reviewer: ```suggestion
**Note**: We include a few tags by default (`service`, `env`, `version`, and `application.id`). For anything else, set the global context using the method above.
```

> **Note**: We include a few tags by default (`service`, `env`, `version`, and `application.id`). For anything else, set the global context using the method above.

Reviewer: ```suggestion
**Note**: A few tags are included by default (`service`, `env`, `version`, and `application.id`). For anything else, set the global context using the method above.
```
